solidity/contracts/AcreMultiAssetVault.sol

```diff
@@ -0,0 +1,278 @@
+        uint256 amount
+    );
+
+    /// @notice Emitted when a deposit is withdrawn.
+    event DepositWithdrawn(
+        address indexed depositOwner,
+        address indexed asset,
+        uint256 indexed depositId,
+        uint256 amount,
+        address receiver
+    );
+
+    /// @notice Details of a deposit.
+    struct DepositInfo {
+        uint256 balance;
+        uint256 mezoDepositId;
+    }
+
+    /// @notice The number of deposits created. Includes the deposits that
+    ///         were fully withdrawn. This is also the identifier of the most
+    ///         recently created deposit.
+    uint256 public depositCount;
+
+    mapping(address => bool) public supportedAssets;
+
+    /// @notice Mapping of the details of deposits:
+    ///         deposit owner -> asset -> deposit id -> DepositInfo
+    /// @dev The deposit ID is unique for each deposit.
+    /// @dev When a deposit is withdrawn, the DepositInfo is deleted.
+    mapping(address => mapping(address => mapping(uint256 => DepositInfo)))
+        public deposits;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _owner,
+        address _mezoPortal,
+        address[] memory _supportedAssets
+    ) public initializer {
+        __Ownable2Step_init();
+        __Ownable_init(_owner);
+
+        if (_mezoPortal == address(0)) {
+            revert ZeroAddress();
+        }
+
+        for (uint256 i = 0; i < _supportedAssets.length; i++) {
+            _addSupportedAsset(_supportedAssets[i]);
+        }
+
+        mezoPortal = IMezoPortal(_mezoPortal);
+    }
+
+    /// @notice Adds an asset to the list of assets supported by the vault.
+    /// @dev This function can only be called by the owner.
+    /// @param asset Address of the asset to be added to the supported assets.
+    function addSupportedAsset(address asset) external onlyOwner {
+        return _addSupportedAsset(asset);
+    }
+
+    /// @notice Adds an asset to the list of assets supported by the vault.
+    /// @param asset Address of the asset to be added to the supported assets.
+    function _addSupportedAsset(address asset) internal {
+        if (asset == address(0)) {
+            revert ZeroAddress();
+        }
+        if (supportedAssets[asset]) {
+            revert AssetAlreadySupported();
+        }
+
+        emit SupportedAssetAdded(asset);
+
+        supportedAssets[asset] = true;
+    }
+
+    /// @notice Removes an asset from the list of assets supported by the vault.
+    /// @dev This function can only be called by the owner.
+    /// @param asset Address of the asset to be removed from the supported assets.
+    function removeSupportedAsset(address asset) external onlyOwner {
+        if (!supportedAssets[asset]) {
+            revert AssetNotSupported();
+        }
+
+        emit SupportedAssetRemoved(asset);
+
+        supportedAssets[asset] = false;
+    }
```

Comment:
We decided to not allow removing assets in Mezo to avoid complicating the business logic - do you expect to use this function? Why would you add and then remove the asset? What about the users that already deposited that asset?

Reply:
If the asset is removed new deposits can't be created for the given asset.
But existing depositors are able to withdraw, since we're not checking if the asset is supported in the withdraw function.

I was curious why Mezo Portal's withdraw function checks if the asset is supported?

Reply:
> why Mezo Portal's withdraw function checks if the asset is supported

I think it was just for consistency.